src/hooks/useInlineErrors.ts

```diff
@@ -21,7 +21,8 @@ function useInlineErrors(
   const [inlineErrors, setInlineErrors] = useState<InlineError[]>([]);
   const setError = useCallback(
     (key: InlineError["key"], error: InlineError["error"]) => {
-      setInlineErrors((inlineErrors) => {
+      setInlineErrors((prevInlineErrors) => {
+        const inlineErrors = [...prevInlineErrors];
```

Comment:
I think, without this fix, the Connection error won't appear straight away in the `WebCLI`, as we were mutating the state here directly.

Reply:
I'd always understood updater functions to be safe as they do not mutate the current state, but rather the functions get queued to be reduced on the next render. However, I couldn't see anywhere in the documentation where it mentions whether it is safe to mutate the previous state inside the function.

I dug into the source and it looks like the updater function is just a reducer function passed to useReducer and in the docs for useReducer it says to use a new object so it doesn't have the same reference.

Saying all that, was there a specific bug that you were running into? I tried both update methods and the render order appeared to be working correctly for me.

Reply:
Thanks for the detailed explanation! After today's discussion, I looked over the code a bit and I think I'll use `immer` here. This is mainly cause, in some cases, we might make a shallow copy here (e.g. when the error has type `Iterable<ReactNode>`), which might prove to have the same issue we are trying to fix right now.